eventsourcing/src/test/java/org/axonframework/eventsourcing/EventCountSnapshotTriggerDefinitionTest.java

```diff
@@ -84,10 +103,26 @@ void testSnapshotterTriggeredOnUnitOfWorkCommit() {
         verify(mockSnapshotter).scheduleSnapshot(aggregate.rootType(), aggregateIdentifier);
     }
 
+    @Test
+    void testSnapshotterTriggeredOnUnitOfWorkRollback() {
```

Comment:
Nit: wouldn't it be beneficial to state this should only occur if the snapshot is triggered in the pre-apply phase? Thus, prior before the aggregate has been initialized? Because, only then will we we invoke this `onCleanUp`

Reply:
Does make the `trigger.initializationFinished();` weird for me right now...so might be have a misconception here. Please tell me if I am wrong on the matter.

Reply:
Alright, I've only just now noticed the verify twice expect it `never()` to happen. That was what I was missing here. No stress!